lib/hirefire/macro/sidekiq.rb

```diff
@@ -28,34 +28,57 @@ def queue(*queues)
           options = {}
         end
 
-        queues = queues.map(&:to_s)
-        queues = ::Sidekiq::Stats.new.queues.map { |name, _| name } if queues.empty?
+        queues.map!(&:to_s)
+        all_queue_names = ::Sidekiq::Queue.all.map(&:name)
 
-        in_queues = queues.inject(0) do |memo, name|
-          memo += ::Sidekiq::Queue.new(name).size
-          memo
-        end
+        # When no queue names are provided (or all of them are), we know we
+        # can peform much faster counts using Sidekiq::Stats and Redis
+        fast_scan = queues.empty? || queues.sort == all_queue_names.sort
 
-        if !options[:skip_scheduled]
-          max = options[:max_scheduled]
-          in_schedule = ::Sidekiq::ScheduledSet.new.inject(0) do |memo, job|
-            memo += 1 if queues.include?(job["queue"]) && job.at <= Time.now
-            break memo if max && memo >= max
-            memo
+        queues = all_queue_names if queues.empty?
+
+        if fast_scan
+          stats = ::Sidekiq::Stats.new
+
+          in_queues = stats.enqueued
+
+          if !options[:skip_scheduled]
+            in_schedule = ::Sidekiq.redis { |c| c.zcount('schedule', '-inf', Time.now.to_f) }
```

Comment:
Is it desirable to respect `max_scheduled` here or is it purely for performance reasons?

Reply:
`max_scheduled` is purely for performance reasons. It allows the user to break out early in the event that further looping would have zero effect on (up-)scaling decisions (as configured in the HireFire UI). So I think you could leave it out if performance isn't an issue with checking the scheduled set in the event that you check all of them anyway.